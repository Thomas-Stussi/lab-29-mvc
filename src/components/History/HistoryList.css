.HistoryList{
  border: 1px solid black;
  height: 92%;
  margin: 2.5px;
  list-style-type: none;
  padding-right: 15px;
  width: 50vh;
  overflow: auto;
}
