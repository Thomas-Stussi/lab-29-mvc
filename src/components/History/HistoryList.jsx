/* eslint-disable max-len */
import React from 'react';
import PropTypes from 'prop-types';
import HistoryItem from './HistoryItem';

const HistoryList = ({ histories }) => {
  const historyElements = histories.map(history => (
    <li key={`${history.method}+${history.url}`}>
      <HistoryItem method={history.method} url={history.url} body={history.body} />
    </li>
  ));

  return (
    <ul>
      {historyElements}
    </ul>
  );
};

HistoryList.propTypes = {
  histories: PropTypes.arrayOf(PropTypes.shape({
    url: PropTypes.string.isRequired,
    body: PropTypes.string,
    method: PropTypes.string.isRequired
  })).isRequired
};

export default HistoryList;
