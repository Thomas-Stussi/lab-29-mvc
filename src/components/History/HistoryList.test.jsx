/* eslint-disable max-len */
import React from 'react';
import { render, cleanup } from '@testing-library/react';
import HistoryList from './HistoryList';

describe('HistoryList component', () => {
  afterEach(() => cleanup());

  const testArray = [
    { url: 'https://www.thecocktaildb.com/api/json/v1/1/random.php', method: 'GET' },
    { url: 'https://www.thecocktaildb.com/api/json/v1/1/random.php', method: 'POST' },
    { url: 'futuramaapi.herokuapp.com/api/quotes', method: 'GET' }
  ];

  it('render HistoryList component', () => {
    const { asFragment } = render(<HistoryList
      histories={testArray}
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});
