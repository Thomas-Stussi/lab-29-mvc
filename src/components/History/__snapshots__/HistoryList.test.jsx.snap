// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistoryList component render HistoryList component 1`] = `
<DocumentFragment>
  <ul>
    <li>
      <h3>
        GET
      </h3>
      <p>
        https://www.thecocktaildb.com/api/json/v1/1/random.php
      </p>
      <p />
    </li>
    <li>
      <h3>
        POST
      </h3>
      <p>
        https://www.thecocktaildb.com/api/json/v1/1/random.php
      </p>
      <p />
    </li>
    <li>
      <h3>
        GET
      </h3>
      <p>
        futuramaapi.herokuapp.com/api/quotes
      </p>
      <p />
    </li>
  </ul>
</DocumentFragment>
`;
